### DISCUSSION TITLE
Example for iframes or flask integrationt o make own apps

### DISCUSSION THREAD

**Discussion by ozgurgunesulkesi** [cockpit-project/cockpit#19182] [Q&A]
Hi,  I am now working with cockpit for a time. But actually not the pro in javascript or html. I am more focused on python. So would be nice if you would give a guide on integrating either iframes or a flask app into cockpit. Because actually I struggle with CSP issue. Actually i am stucked in the manifest.json configuration

**Comment by martinpitt** [comment]
There is an [example](https://github.com/cockpit-project/cockpit/blob/main/examples/integrate-terminal/integrate-terminal.html) for embedding a cockpit component into another web app. But I suppose you actually want it the other way around, show another web app inside a cockpit page.

This only works if both cockpit's web server and that other web app come from the **same origin**. In other words, you need to have a reverse proxy somewhere.

Please provide more details about your manifest, CSP error messages, network setup, etc. This is too vague to be able to help you, sorry.
